WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyRingViewModel.cs

```diff
@@ -39,53 +40,60 @@ public PrivacyRingViewModel(WalletViewModel walletViewModel)
 
 		PreviewItems.Add(PrivacyTile);
 
-		_itemsSourceList
+		SetupCancel(enableCancel: true, enableCancelOnEscape: true, enableCancelOnPressed: true);
+	}
+
+	public PrivacyControlTileViewModel PrivacyTile { get; }
+
+	public ObservableCollectionExtended<PrivacyRingItemViewModel> Items { get; } = new();
+	public ObservableCollectionExtended<IPrivacyRingPreviewItem> PreviewItems { get; } = new();
+
+	public Wallet Wallet { get; }
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		var itemsSourceList = new SourceList<PrivacyRingItemViewModel>();
+
+		itemsSourceList
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Bind(Items)
 			.DisposeMany()
-			.Subscribe();
-
-		_coinsUpdated =
-			walletViewModel.UiTriggers.PrivacyProgressUpdateTrigger
-						  .Merge(walletViewModel
-						  .WhenAnyValue(w => w.IsCoinJoining)
-						  .ToSignal());
-
-		_coinsUpdated
-			.Select(_ => walletViewModel.Wallet.GetPockets())
-			.ObserveOn(RxApp.MainThreadScheduler)
-			.Subscribe(RefreshCoinsList);
-
-		_disposables.Add(_itemsSourceList);
+			.Subscribe()
+			.DisposeWith(disposables);
+
+		_walletViewModel.UiTriggers
+						.PrivacyProgressUpdateTrigger
+						.Merge(_walletViewModel.WhenAnyValue(w => w.IsCoinJoining).ToSignal())
```

Comment:
Why should it be redrawn when the coinjoin state is changing?

Reply:
You're right. I completely glossed over that when looking at the code. It was a leftover from a previous design.

I wanted to highlight coins that are currently under coinjoin, but I will do that in a different way anyways. 

latest commit removes that trigger.